internal/credential/repository_credentiallibrary.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// A CredentialLibraryRepository provides read only information
+// generic to all credential store types.
+type CredentialLibraryRepository struct {
```

Comment:
Why `CredentialLibraryRepository` and not just `Repository`?

Reply:
Because there will be a `CredentialStoreRepository` and `CredentialRepository` alongside this soon. They all have to answer the question "How many of X are there?" independently, so I thought naming the repository was better than having `GetTotalCredentialStores()`, `GetTotalCredentialLibraries()` on a single meta-repository. What do you think?

Reply:
It's more consistent with other repositories, which generally deal with a single resource type (e.g. ssh, tcp, oidc, password).

Reply:
Here's the credential store implementation: https://github.com/hashicorp/boundary/pull/3742

Reply:
And credentials: https://github.com/hashicorp/boundary/pull/3747